assets/src/edit-story/components/devTools/devTools.js

```diff
@@ -0,0 +1,129 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import { useRef, useState } from 'react';
+import styled from 'styled-components';
+
+/**
+ * Internal dependencies
+ */
+import { useGlobalKeyDownEffect } from '../keyboard';
+import { useStory } from '../../app/story';
+import { useSnackbar } from '../../app/snackbar';
+
+const Container = styled.div`
+  position: fixed;
+  z-index: 2147483646;
+  bottom: 0;
+  left: 0;
+  display: flex;
+  flex-direction: column;
+  width: 35%;
+  height: 50%;
+  border: 6px solid ${({ theme }) => theme.colors.bg.v4};
+  background: ${({ theme }) => theme.colors.bg.v1};
+  color: ${({ theme }) => theme.colors.fg.v2};
+`;
+
+const Textarea = styled.textarea`
+  flex: 1;
+  border: 0;
+  background: ${({ theme }) => theme.colors.bg.v1};
+  color: ${({ theme }) => theme.colors.fg.v2};
+`;
+
+const Options = styled.div`
+  display: flex;
+  justify-content: space-between;
+  padding: 6px;
+`;
+
+const Label = styled.label`
+  margin-right: 10px;
+`;
+
+function DevTools() {
+  const [isVisible, setIsVisible] = useState(false);
+  const [isBase64, setIsBase64] = useState(false);
+  const [isWholeState, setWholeState] = useState(false);
+  const { state } = useStory();
+  const {
+    currentPageIndex,
+    hasSelection,
+    selectedElementIds,
+    story: { stylePresets },
+    pages,
+  } = state;
+  const storyData = isWholeState
+    ? state
+    : {
+        currentPageIndex,
+        hasSelection,
+        selectedElementIds,
+        story: {
+          stylePresets,
+        },
+        pages,
+      };
+  const { showSnackbar } = useSnackbar();
+  const textareaRef = useRef();
+
+  const toggleWholeState = () => setWholeState((v) => !v);
+  const toggleBase64 = () => setIsBase64((v) => !v);
+  const toggleVisible = () => setIsVisible((v) => !v);
+  const copyToClipboard = () => {
+    textareaRef.current.select();
+    document.execCommand('copy');
+    textareaRef.current.setSelectionRange(0, 0);
+    showSnackbar({ message: 'Copied to clipboard', timeout: 1200 });
+  };
+
+  useGlobalKeyDownEffect(
+    { key: 'mod+shift+alt+j', editable: true },
+    toggleVisible
+  );
+
+  const jsonStr = JSON.stringify(storyData, null, 2);
+  const output = isBase64 ? window.btoa(jsonStr) : jsonStr;
+
+  return isVisible ? (
+    <Container>
+      <Options>
+        <div>
+          <Label>
+            <input type="checkbox" checked={isBase64} onChange={toggleBase64} />
+            {'Base64'}
+          </Label>
+          <Label>
+            <input
+              type="checkbox"
+              checked={isWholeState}
+              onChange={toggleWholeState}
+            />
+            {'Whole state'}
+          </Label>
+        </div>
+        <button onClick={copyToClipboard}>{'Copy'}</button>
+      </Options>
+      <Textarea ref={textareaRef} className="mousetrap" value={output} />
```

Comment:
Why is className "mousetrap"?

Reply:
So you can close the DevTools with shortcut after you copy the contents (or in general when this textarea is focused).
I will also add the X button there.

> By default all keyboard events will not fire if you are inside of a textarea, input, or select to prevent undesirable things from happening.
> If you want them to fire you can add the class mousetrap to the element.

We can also handle that in [`stopCallback`](https://craig.is/killing/mice) so you can trigger DevTools from anywhere (also inputs).

Reply:
TIL. But I'd rather be explicit. If you're using our keyboard handlers, you should be able to use `{editable: true}`.

Reply:
```js
{ key: 'mod+shift+alt+j', editable: true }
```
is already there, but will not handle all cases.